src/features/payment/Payment.js

```diff
@@ -20,86 +20,52 @@ class CheckoutContainer extends React.Component {
-    this.paymentComponent = null;
 
-    this.onSubmit = this.onSubmit.bind(this);
-    this.onAdditionalDetails = this.onAdditionalDetails.bind(this);
     this.processPaymentResponse = this.processPaymentResponse.bind(this);
   }
 
   componentDidMount() {
-    this.props.getPaymentMethods();
+    this.props.initiateCheckout(this.props.type);
   }
 
   componentDidUpdate(prevProps) {
-    const { paymentMethodsRes: paymentMethodsResponse, config, paymentRes, paymentDetailsRes, error } = this.props.payment;
+    const { sessionAndOrderRef, config, error } = this.props.payment;
     if (error && error !== prevProps.payment.error) {
       window.location.href = `/status/error?reason=${error}`;
       return;
     }
-    if (
-      paymentMethodsResponse &&
-      config &&
-      (paymentMethodsResponse !== prevProps.payment.paymentMethodsRes || config !== prevProps.payment.config)
-    ) {
-      // @ts-ignore
-      // eslint-disable-next-line no-undef
-      this.checkout = new AdyenCheckout({
-        ...config,
-        paymentMethodsResponse,
-        onAdditionalDetails: this.onAdditionalDetails,
-        onSubmit: this.onSubmit,
-      });
 
-      this.checkout.create(this.props.type).mount(this.paymentContainer.current);
+    const configWithSession = {
+      ...config,
+      session : sessionAndOrderRef[0],
+      onPaymentCompleted : ((res, _) => {console.log("payment completed " + res); this.processPaymentResponse(res);}),
+      onError : ((err, _) => {console.log("payment error " + err); ; this.processPaymentResponse(err);}),      
     }
-    if (paymentRes && paymentRes !== prevProps.payment.paymentRes) {
-      this.processPaymentResponse(paymentRes[0]); // get only response from tuple
-    }
-    if (paymentRes && paymentDetailsRes !== prevProps.payment.paymentDetailsRes) {
-      this.processPaymentResponse(paymentDetailsRes);
-    }
-  }
 
-  processPaymentResponse(paymentRes) {
-    if (paymentRes.action) {
-      this.paymentComponent.handleAction(paymentRes.action);
-    } else {
-      switch (paymentRes.resultCode) {
-        case "Authorised":
-          window.location.href = "/status/success";
-          break;
-        case "Pending":
-        case "Received":
-          window.location.href = "/status/pending";
-          break;
-        case "Refused":
-          window.location.href = "/status/failed";
-          break;
-        default:
-          window.location.href = "/status/error";
-          break;
-      }
-    }
+    // @ts-ignore
+    // eslint-disable-next-line no-undef
+    this.checkout = new AdyenCheckout(configWithSession)
+      .then(checkout => { checkout.create(this.props.type).mount(this.paymentContainer.current);});
   }
 
-  onSubmit(state, component) {
-    if (state.isValid) {
-      this.props.initiatePayment({
-        ...state.data,
-        origin: window.location.origin,
-      });
-      this.paymentComponent = component;
+  processPaymentResponse(paymentRes) {
```

Comment:
You could use a common redirect component which renders a `<Redirect/>` based on a resultCode (Authorized/Pending/Received/..)

Reply:
Hum, I'm not sure I understand what you mean. It's already how it's being done currently? 

        <Route path="/status/:type">
          <Status />
        </Route>

Reply:
There is some duplication in processPaymentResponse, I think it would be better if this duplicate logic would be moved to a new component.

Reply:
Indeed. That would be picked up as part of https://github.com/adyen-examples/adyen-react-online-payments/issues/9 I expect?